subsys/shell/shell_bt_nus.c

```diff
@@ -105,6 +115,9 @@ static int enable(const struct shell_transport *transport, bool blocking_tx)
 		return -ENOTSUP;
 	}
 
+	LOG_DBG("Waiting for the NUS notification to be enabled");
+	k_sem_take(&shell_bt_nus_ready, K_FOREVER);
```

Comment:
Is this really wise? Blocking forever in `enable`?

Does the application handle when connections are disconnected, or unsubscribes?

Reply:
Agree with @joerchan, this seems a bit fragile. If this requires an actual BLE connection, we cannot block forever unless we have a guarantee that, on failure or timeout, this semaphore will be posted somehow.

Reply:
I think this can be used here because enabling a shell when transport isn't ready doesn't make much sense. I added additional `k_sem_give` function to API disable function and `k_sem_reset` for API enabled function in case of the disconnection event. On the unsubscribe event the shell will return an error when it attempts to send any data to terminal I didn't change a behavior here. At the moment, I have no better idea on how to synchronize it. Maybe @nordic-krch could suggest something ?

Reply:
what about removing this semaphore and changing early return from `write`? Now, write returns immediately if `bt_nus->ctrl_blk->conn == NULL`. This could be changed to return if nus notifications are disabled. Would that make more sense?

Reply:
talked with @KAGA164 offline, i think that semaphore is the cleanest approach here.